// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.comp5325.assignment4";
option java_outer_classname = "COMP5325Bank";
option objc_class_prefix = "CR";

package ClientRequest;

// The greeting service definition.
service Greeter {
  // Sends a action
  rpc GetBalance (ClientRequest) returns (ClientResponse) {}
  rpc Withdraw (ClientRequest) returns (ClientResponse) {}
  rpc Deposit (ClientRequest) returns (ClientResponse) {}
}

// The request message containing the client request
message ClientRequest {
  int32 actionType = 1;
  int32 actionId = 2;
  string acctId = 3;
  int32 requestAmt = 4;
}

// The response message containing the client request
message ClientResponse {
  string status = 1;
  int32 actionId = 2;
  string acctId = 3;
  int32 responseAmt = 4;
}
